service/config.go

```diff
@@ -32,10 +45,52 @@ func (s *configService) SetJWTSecret(secret []byte) error {
 	return s.repo.SetJWTSecret(secret)
 }
 
-// SetSuperAdmin will first make sure the provided username is a valid account
-// and then assign it as the SuperAdmin role. Only one is allowed to be set at a
-// time.
-func (s *configService) SetSuperAdmin(name string) error {
-	// TODO: Make sure this is a valid user first.
-	return s.repo.SetSuperAdmin(name)
+// CreateSuperAdmin will take a username, generate a new password, and save this
+// user with SuperAdmin permissions.
+// TODO: Refactor so we don't return a full user and pass?
+func (s *configService) CreateSuperAdmin(name string) (*model.User, string, error) {
+	sa, err := s.repo.SuperAdmin()
+	if err != nil {
+		return nil, "", err
+	}
+	if sa != "" {
+		return nil, "", ErrSuperAdminExists
+	}
+
+	p, err := newPassword()
+	if err != nil {
+		return nil, "", err
+	}
+
+	u, err := model.NewUser(name, p, auth.RoleSuperAdmin)
+	if err != nil {
+		return nil, "", err
+	}
+
+	err = s.userService.SaveUser(u)
+	if err != nil {
+		return nil, "", err
+	}
+
+	err = s.repo.SetSuperAdmin(name)
```

Comment:
Question might just be cuz I don't quite understand the backend arch yet but if save user fails with an error are we stuck in a place where we made a user but haven't set super admin permissions to that user?

Reply:
If SaveUser fails, the user is just never created really. Although if SetSuperAdmin fails, the User will still exist and have been saved so I should fix that.